Apps/Src/SendTritium.c

```diff
@@ -0,0 +1,621 @@
+
+    if(Minion_Read_Pin(CRUZ_EN, &err) && cruiseEnableCounter < DEBOUNCE_PERIOD){cruiseEnableCounter++;}
+    else if(cruiseEnableCounter > 0){cruiseEnableCounter--;}
+
+    if(Minion_Read_Pin(CRUZ_ST, &err) && cruiseSetCounter < DEBOUNCE_PERIOD){cruiseSetCounter++;}
+    else if(cruiseSetCounter > 0){cruiseSetCounter--;}
+    
+    // Update gears
+    bool forwardSwitch = Minion_Read_Pin(FOR_SW, &err);
+    bool reverseSwitch = Minion_Read_Pin(REV_SW, &err);
+    bool forwardGear = (forwardSwitch && !reverseSwitch);
+    bool reverseGear = (!forwardSwitch && reverseSwitch);
+    bool neutralGear = (!forwardSwitch && !reverseSwitch);
+
+    uint8_t gearFault = (uint8_t) forwardGear + (uint8_t) reverseGear + (uint8_t) neutralGear;
+    static uint8_t gearFaultCnt = 0;
+
+    if(gearFault != 1){
+        // Fault behavior
+        if(gearFaultCnt > GEAR_FAULT_THRESHOLD) state = FSM[NEUTRAL_DRIVE];
+        else gearFaultCnt++;
+    }
+    else{
+        gearFaultCnt = 0;
+    }
+
+    if(neutralGear) gear = NEUTRAL_GEAR;
+    else if(forwardGear) gear = FORWARD_GEAR;
+    else if(reverseGear) gear = REVERSE_GEAR;
+    else gear = NEUTRAL_GEAR;
+
+    // Debouncing
+    if(onePedalCounter == DEBOUNCE_PERIOD){onePedalButton = true;}
+    else if(onePedalCounter == 0){onePedalButton = false;}
+
+    if(cruiseEnableCounter == DEBOUNCE_PERIOD){cruiseEnableButton = true;}
+    else if(cruiseEnableCounter == 0){cruiseEnableButton = false;}
+
+    if(cruiseSetCounter == DEBOUNCE_PERIOD){cruiseSet = true;}
+    else if(cruiseSetCounter == 0){cruiseSet = false;}
+
+    // Toggle
+    if(onePedalButton != onePedalPrevious && onePedalPrevious){onePedalEnable = !onePedalEnable;}
+    if(!regenEnable) onePedalEnable = false;
+    onePedalPrevious = onePedalButton;
+    
+    if(cruiseEnableButton != cruiseEnablePrevious && cruiseEnablePrevious){cruiseEnable = !cruiseEnable;}
+    cruiseEnablePrevious = cruiseEnableButton;
+    
+    // Get observed velocity
+    velocityObserved = Motor_Velocity_Get();
+}
+#endif
+
+// Helper Functions
+
+/**
+ * @brief Converts integer percentage to float percentage
+ * @param percent integer percentage from 0-100
+ * @returns float percentage from 0.0-1.0
+*/
+extern const float pedalToPercent[];
+static float percentToFloat(uint8_t percent){
+    if(percent > 100){
+        return 1.0f;
+    }
+    return pedalToPercent[percent];
+}
+
+/**
+ * @brief Linearly map range of integers to another range of integers. 
+ * in_min to in_max is mapped to out_min to out_max.
+ * @param input input integer value
+ * @param in_min minimum value of input range
+ * @param in_max maximum value of input range
+ * @param out_min minimum value of output range
+ * @param out_max maximum value of output range 
+ * @returns integer value from out_min to out_max
+*/
+static uint8_t map(uint8_t input, uint8_t in_min, uint8_t in_max, uint8_t out_min, uint8_t out_max){
+    if(in_max >= in_min) in_max = in_min;
```

Comment:
Why is it checking that `in_max >= in_min` and then setting them equal like that? If `in_max` is larger, shouldn't it remain larger?